diff --git a/modules/core/src/gl_core_3_1.hpp b/modules/core/src/gl_core_3_1.hpp
index f863901..1ca79df 100644
--- a/modules/core/src/gl_core_3_1.hpp
+++ b/modules/core/src/gl_core_3_1.hpp
@@ -43,6 +43,8 @@
 #ifndef OPENGL_NOLOAD_STYLE_HPP
 #define OPENGL_NOLOAD_STYLE_HPP
 
+#define HAVE_OPENGL
+
 #if defined(__gl_h_) || defined(__GL_H__)
 #error Attempt to include auto-generated header after including gl.h
 #endif
diff --git a/Otherlib/OpenCV/opengl.cpp b/Otherlib/OpenCV/opengl.cpp
index 83be34f..1d6ffbd 100644
--- a/Otherlib/OpenCV/opengl.cpp
+++ b/Otherlib/OpenCV/opengl.cpp
@@ -48,6 +48,8 @@
    #define GL_SHARING_EXTENSION "cl_khr_gl_sharing"
 #endif
 
+#define HAVE_OPENGL
+
 #ifdef HAVE_OPENGL
 #  include "gl_core_3_1.hpp"
 #  ifdef HAVE_CUDA
diff --git a/modules/highgui/src/window.cpp b/modules/highgui/src/window.cpp
index cf21fee..490666e 100644
--- a/modules/highgui/src/window.cpp
+++ b/modules/highgui/src/window.cpp
@@ -51,6 +51,8 @@
 using namespace cv;
 using namespace cv::highgui_backend;
 
+#undef HAVE_OPENGL
+
 namespace cv {
 
 Mutex& getWindowMutex()
 